xclim/indices/_conversion.py

```diff
@@ -895,6 +909,115 @@ def rain_approximation(
     return prra
 
 
+@declare_units(snw="[mass]/[area]", snr="[mass]/[volume]")
+def snow_depth_approximation(
+    snw: xr.DataArray,
+    snr: xr.DataArray,
+) -> xr.DataArray:
+    """Approximation of snow depth from snow amount and density.
+
+    Parameters
+    ----------
+    snw : xr.DataArray
+        Snow amount [kg/m^2].
+        If snow water equivalent (`swe` [m]) is provided instead, will be converted to `snw` before calculating.
+    snr : xr.DataArray, optional
+        Snow density [kg/m^3].
+
+    Returns
+    -------
+    xr.DataArray, [m]
+        Snow depth.
+    """
+    snw = convert_units_to(snw, "kg m-2", context="hydro")
```

Comment:
```suggestion
    snw = convert_units_to(snw, "kg m-2")
```
I think I was the one to suggest to add the hydro context, but I'm not sure anymore...

The point of this function is that snow has a variable density, whereas the `hydro` context assumed a 1000 kg/m³. This would only be useful in the case where `swe` is given, but one should really call an explicit `convert_units_to`  in that case, no?

Reply:
Also, I now realize that the  `declare_units` will make fail any attempt to pass `snw` in a dimensionality other than "[mass]/[area]". Thus, `context='hydro'` is never used anyway.